src/ui/common/src/components/LogViewer/index.tsx

```diff
@@ -0,0 +1,70 @@
+import { TabContext, TabList, TabPanel } from '@mui/lab';
+import { Tab } from '@mui/material';
+import Box from '@mui/material/Box';
+import React, { useState } from 'react';
+
+import { Error, Logs } from '../../utils/shared';
+
+type Props = {
+  logs?: Logs;
+  err?: Error;
+  contentHeight?: string;
+};
+const LogViewer: React.FC<Props> = ({ logs, err, contentHeight = '10vh' }) => {
+  const [currentTab, setCurrentTab] = useState(err ? '3' : '1');
+
+  const tabPanelOptions = {
+    height: contentHeight,
+    overflow: 'scroll',
+    fontFamily: 'monospace, monospace',
+  };
+  const errorTabPanelOptions = { ...tabPanelOptions, color: 'red.500' };
+  const empty = { color: 'gray.200' };
+
+  const hasOutput = (obj) => {
+    return obj !== undefined && obj.length > 0;
+  };
+
+  return (
+    <Box sx={{ mb: 4 }} pb={1}>
+      <TabContext value={currentTab}>
+        <Box sx={{ borderBottom: 1, borderColor: 'divider' }}>
+          <TabList
+            onChange={(_, tab) => setCurrentTab(tab)}
+            aria-label="log viewer"
+          >
+            <Tab
+              sx={hasOutput(logs?.stdout) ? {} : empty}
+              label="Stdout"
+              value="1"
+            />
+            <Tab
+              sx={hasOutput(logs?.stderr) ? {} : empty}
+              label="Stderr"
+              value="2"
+            />
+            <Tab
+              sx={err !== undefined && hasOutput(err?.tip) ? {} : empty}
+              label="Errors"
+              value="3"
+            />
+          </TabList>
+        </Box>
+
+        <TabPanel sx={tabPanelOptions} value="1">
+          {hasOutput(logs?.stdout) ? logs.stdout : 'No output to display.'}
```

Comment:
Bit of a nitpick, but could you make 'No output to display.' a const please?
```
const noOutputMessage = 'No output to display.'
```

Same for the no errors message and no logs message